V2/gameFuseChat.js

```diff
@@ -0,0 +1,195 @@
+class GameFuseChat {
+    constructor(id, participants, messages){
+        this.id = id
+        this.participants = participants; // participants must be before messages so that the messages can access the participant user data in the UserCache
+        this.messages = messages;
+    }
+
+    getID() {
+        return this.id;
+    }
+
+    getMessages() {
+        return this.messages;
+    }
+
+    getParticipants() {
+        return this.participants;
+    }
+
+    // recipients => array of user objects with whom the chat should be.
+    // Can also just be a user object.
+    // Can also be a clan. Or some other type of "group"
+    static async sendMessage(recipients, firstMessage, callback = undefined){
+        try {
+            recipients = Array.isArray(recipients) ? recipients : [recipients]
+
+            // TODO: should we allow all these different ways to pass the recipients? Or should we force them to send it as one specific way?
```

Comment:
Resolve this TODO before merging. Currently we are only using the username way (also see the instance method sendMessage on gameFuseUser, which extracts the username and sends it to this method).